[Benchmark][Windows] One or more tests failed while running tests from 'System.Net.Http.Performance.Tests'

The specified benchmark has been broken for few days now, and it's blocking uploading performance results on Windows.

System.Net.Http.Tests.SocketsHttpHandlerPerfTest.Get(ssl: True, connectionPerRequest: True, chunkedResponse: False, responseLength: 1): Only one usage of each socket address (protocol/network address/port) is normally permitted

```log
[4/23/2018 1:30:50 PM][INF] Running 16 [Benchmark]s
  [4/23/2018 1:30:50 PM][INF]   System.Net.Http.Tests.SocketsHttpHandlerPerfTest.Get(ssl: False, connectionPerRequest: True, chunkedResponse: False, responseLength: 100000)
  [4/23/2018 1:30:56 PM][INF]   System.Net.Http.Tests.SocketsHttpHandlerPerfTest.Get(ssl: False, connectionPerRequest: False, chunkedResponse: True, responseLength: 1)
  [4/23/2018 1:30:57 PM][INF]   System.Net.Http.Tests.SocketsHttpHandlerPerfTest.Get(ssl: True, connectionPerRequest: False, chunkedResponse: True, responseLength: 1)
  [4/23/2018 1:30:58 PM][INF]   System.Net.Http.Tests.SocketsHttpHandlerPerfTest.Get(ssl: True, connectionPerRequest: True, chunkedResponse: False, responseLength: 1)
  [4/23/2018 1:31:05 PM][ERR] System.Net.Http.Tests.SocketsHttpHandlerPerfTest.Get(ssl: True, connectionPerRequest: True, chunkedResponse: False, responseLength: 1): Only one usage of each socket address (protocol/network address/port) is normally permitted
  [4/23/2018 1:31:05 PM][ERR]    at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken) in C:\J\w\perf_windows_---356c2fc4\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\ConnectHelper.cs:line 86
     at System.Net.Http.HttpConnectionPool.CreateConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken) in C:\J\w\perf_windows_---356c2fc4\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpConnectionPool.cs:line 342
     at System.Net.Http.HttpConnectionPool.WaitForCreatedConnectionAsync(ValueTask`1 creationTask) in C:\J\w\perf_windows_---356c2fc4\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpConnectionPool.cs:line 484
     at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken) in C:\J\w\perf_windows_---356c2fc4\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpConnectionPool.cs:line 275
     at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken) in C:\J\w\perf_windows_---356c2fc4\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\RedirectHandler.cs:line 32
     at System.Net.Http.Tests.SocketsHttpHandlerPerfTest.Get(Boolean ssl, Boolean connectionPerRequest, Boolean chunkedResponse, Int32 responseLength) in C:\J\w\perf_windows_---356c2fc4\src\System.Net.Http\tests\Performance\Perf.SocketsHttpHandler.cs:line 107
     at Microsoft.Xunit.Performance.BenchmarkTestInvoker.<>c__DisplayClass1_0.<<CallTestMethod>b__0>d.MoveNext()
  --- End of stack trace from previous location where exception was thrown ---
     at Microsoft.Xunit.Performance.Internal.BenchmarkIterator.RunAsync(Func`1 testMethod)
     at Xunit.Sdk.TestInvoker`1.<>c__DisplayClass46_1.<<InvokeTestMethodAsync>b__1>d.MoveNext() in C:\BuildAgent\work\cb37e9acf085d108\src\xunit.execution\Sdk\Frameworks\Runners\TestInvoker.cs:line 227
  --- End of stack trace from previous location where exception was thrown ---
     at Xunit.Sdk.ExecutionTimer.AggregateAsync(Func`1 asyncAction) in C:\BuildAgent\work\cb37e9acf085d108\src\xunit.execution\Sdk\Frameworks\ExecutionTimer.cs:line 48
     at Xunit.Sdk.ExceptionAggregator.RunAsync(Func`1 code) in C:\BuildAgent\work\cb37e9acf085d108\src\xunit.core\Sdk\ExceptionAggregator.cs:line 90
  [4/23/2018 1:31:05 PM][INF]   System.Net.Http.Tests.SocketsHttpHandlerPerfTest.Get(ssl: False, connectionPerRequest: True, chunkedResponse: True, responseLength: 100000)
  [4/23/2018 1:31:11 PM][INF]   System.Net.Http.Tests.SocketsHttpHandlerPerfTest.Get(ssl: True, connectionPerRequest: True, chunkedResponse: True, responseLength: 1)
  [4/23/2018 1:31:17 PM][ERR] System.Net.Http.Tests.SocketsHttpHandlerPerfTest.Get(ssl: True, connectionPerRequest: True, chunkedResponse: True, responseLength: 1): Only one usage of each socket address (protocol/network address/port) is normally permitted
  [4/23/2018 1:31:17 PM][ERR]    at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken) in C:\J\w\perf_windows_---356c2fc4\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\ConnectHelper.cs:line 86
     at System.Net.Http.HttpConnectionPool.CreateConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken) in C:\J\w\perf_windows_---356c2fc4\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpConnectionPool.cs:line 342
     at System.Net.Http.HttpConnectionPool.WaitForCreatedConnectionAsync(ValueTask`1 creationTask) in C:\J\w\perf_windows_---356c2fc4\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpConnectionPool.cs:line 484
     at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken) in C:\J\w\perf_windows_---356c2fc4\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpConnectionPool.cs:line 275
     at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken) in C:\J\w\perf_windows_---356c2fc4\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\RedirectHandler.cs:line 32
     at System.Net.Http.Tests.SocketsHttpHandlerPerfTest.Get(Boolean ssl, Boolean connectionPerRequest, Boolean chunkedResponse, Int32 responseLength) in C:\J\w\perf_windows_---356c2fc4\src\System.Net.Http\tests\Performance\Perf.SocketsHttpHandler.cs:line 107
     at Microsoft.Xunit.Performance.BenchmarkTestInvoker.<>c__DisplayClass1_0.<<CallTestMethod>b__0>d.MoveNext()
  --- End of stack trace from previous location where exception was thrown ---
     at Microsoft.Xunit.Performance.Internal.BenchmarkIterator.RunAsync(Func`1 testMethod)
     at Xunit.Sdk.TestInvoker`1.<>c__DisplayClass46_1.<<InvokeTestMethodAsync>b__1>d.MoveNext() in C:\BuildAgent\work\cb37e9acf085d108\src\xunit.execution\Sdk\Frameworks\Runners\TestInvoker.cs:line 227
  --- End of stack trace from previous location where exception was thrown ---
     at Xunit.Sdk.ExecutionTimer.AggregateAsync(Func`1 asyncAction) in C:\BuildAgent\work\cb37e9acf085d108\src\xunit.execution\Sdk\Frameworks\ExecutionTimer.cs:line 48
     at Xunit.Sdk.ExceptionAggregator.RunAsync(Func`1 code) in C:\BuildAgent\work\cb37e9acf085d108\src\xunit.core\Sdk\ExceptionAggregator.cs:line 90
  [4/23/2018 1:31:17 PM][INF]   System.Net.Http.Tests.SocketsHttpHandlerPerfTest.Get(ssl: True, connectionPerRequest: True, chunkedResponse: True, responseLength: 100000)
  [4/23/2018 1:31:29 PM][INF]   System.Net.Http.Tests.SocketsHttpHandlerPerfTest.Get(ssl: False, connectionPerRequest: False, chunkedResponse: False, responseLength: 1)
  [4/23/2018 1:31:29 PM][INF]   System.Net.Http.Tests.SocketsHttpHandlerPerfTest.Get(ssl: True, connectionPerRequest: False, chunkedResponse: False, responseLength: 100000)
  [4/23/2018 1:31:34 PM][INF]   System.Net.Http.Tests.SocketsHttpHandlerPerfTest.Get(ssl: False, connectionPerRequest: False, chunkedResponse: True, responseLength: 100000)
  [4/23/2018 1:31:37 PM][INF]   System.Net.Http.Tests.SocketsHttpHandlerPerfTest.Get(ssl: True, connectionPerRequest: True, chunkedResponse: False, responseLength: 100000)
  [4/23/2018 1:31:39 PM][ERR] System.Net.Http.Tests.SocketsHttpHandlerPerfTest.Get(ssl: True, connectionPerRequest: True, chunkedResponse: False, responseLength: 100000): Only one usage of each socket address (protocol/network address/port) is normally permitted
  [4/23/2018 1:31:39 PM][ERR]    at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken) in C:\J\w\perf_windows_---356c2fc4\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\ConnectHelper.cs:line 86
     at System.Net.Http.HttpConnectionPool.CreateConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken) in C:\J\w\perf_windows_---356c2fc4\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpConnectionPool.cs:line 342
     at System.Net.Http.HttpConnectionPool.WaitForCreatedConnectionAsync(ValueTask`1 creationTask) in C:\J\w\perf_windows_---356c2fc4\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpConnectionPool.cs:line 484
     at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken) in C:\J\w\perf_windows_---356c2fc4\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpConnectionPool.cs:line 275
     at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken) in C:\J\w\perf_windows_---356c2fc4\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\RedirectHandler.cs:line 32
     at System.Net.Http.Tests.SocketsHttpHandlerPerfTest.Get(Boolean ssl, Boolean connectionPerRequest, Boolean chunkedResponse, Int32 responseLength) in C:\J\w\perf_windows_---356c2fc4\src\System.Net.Http\tests\Performance\Perf.SocketsHttpHandler.cs:line 107
     at Microsoft.Xunit.Performance.BenchmarkTestInvoker.<>c__DisplayClass1_0.<<CallTestMethod>b__0>d.MoveNext()
  --- End of stack trace from previous location where exception was thrown ---
     at Microsoft.Xunit.Performance.Internal.BenchmarkIterator.RunAsync(Func`1 testMethod)
     at Xunit.Sdk.TestInvoker`1.<>c__DisplayClass46_1.<<InvokeTestMethodAsync>b__1>d.MoveNext() in C:\BuildAgent\work\cb37e9acf085d108\src\xunit.execution\Sdk\Frameworks\Runners\TestInvoker.cs:line 227
  --- End of stack trace from previous location where exception was thrown ---
     at Xunit.Sdk.ExecutionTimer.AggregateAsync(Func`1 asyncAction) in C:\BuildAgent\work\cb37e9acf085d108\src\xunit.execution\Sdk\Frameworks\ExecutionTimer.cs:line 48
     at Xunit.Sdk.ExceptionAggregator.RunAsync(Func`1 code) in C:\BuildAgent\work\cb37e9acf085d108\src\xunit.core\Sdk\ExceptionAggregator.cs:line 90
  [4/23/2018 1:31:39 PM][INF]   System.Net.Http.Tests.SocketsHttpHandlerPerfTest.Get(ssl: True, connectionPerRequest: False, chunkedResponse: False, responseLength: 1)
  [4/23/2018 1:31:40 PM][INF]   System.Net.Http.Tests.SocketsHttpHandlerPerfTest.Get(ssl: False, connectionPerRequest: True, chunkedResponse: False, responseLength: 1)
  [4/23/2018 1:31:42 PM][ERR] System.Net.Http.Tests.SocketsHttpHandlerPerfTest.Get(ssl: False, connectionPerRequest: True, chunkedResponse: False, responseLength: 1): Only one usage of each socket address (protocol/network address/port) is normally permitted
  [4/23/2018 1:31:42 PM][ERR]    at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken) in C:\J\w\perf_windows_---356c2fc4\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\ConnectHelper.cs:line 86
     at System.Net.Http.HttpConnectionPool.CreateConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken) in C:\J\w\perf_windows_---356c2fc4\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpConnectionPool.cs:line 342
     at System.Net.Http.HttpConnectionPool.WaitForCreatedConnectionAsync(ValueTask`1 creationTask) in C:\J\w\perf_windows_---356c2fc4\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpConnectionPool.cs:line 484
     at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken) in C:\J\w\perf_windows_---356c2fc4\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpConnectionPool.cs:line 275
     at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken) in C:\J\w\perf_windows_---356c2fc4\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\RedirectHandler.cs:line 32
     at System.Net.Http.Tests.SocketsHttpHandlerPerfTest.Get(Boolean ssl, Boolean connectionPerRequest, Boolean chunkedResponse, Int32 responseLength) in C:\J\w\perf_windows_---356c2fc4\src\System.Net.Http\tests\Performance\Perf.SocketsHttpHandler.cs:line 107
     at Microsoft.Xunit.Performance.BenchmarkTestInvoker.<>c__DisplayClass1_0.<<CallTestMethod>b__0>d.MoveNext()
  --- End of stack trace from previous location where exception was thrown ---
     at Microsoft.Xunit.Performance.Internal.BenchmarkIterator.RunAsync(Func`1 testMethod)
     at Xunit.Sdk.TestInvoker`1.<>c__DisplayClass46_1.<<InvokeTestMethodAsync>b__1>d.MoveNext() in C:\BuildAgent\work\cb37e9acf085d108\src\xunit.execution\Sdk\Frameworks\Runners\TestInvoker.cs:line 227
  --- End of stack trace from previous location where exception was thrown ---
     at Xunit.Sdk.ExecutionTimer.AggregateAsync(Func`1 asyncAction) in C:\BuildAgent\work\cb37e9acf085d108\src\xunit.execution\Sdk\Frameworks\ExecutionTimer.cs:line 48
     at Xunit.Sdk.ExceptionAggregator.RunAsync(Func`1 code) in C:\BuildAgent\work\cb37e9acf085d108\src\xunit.core\Sdk\ExceptionAggregator.cs:line 90
  [4/23/2018 1:31:42 PM][INF]   System.Net.Http.Tests.SocketsHttpHandlerPerfTest.Get(ssl: False, connectionPerRequest: False, chunkedResponse: False, responseLength: 100000)
  [4/23/2018 1:31:45 PM][INF]   System.Net.Http.Tests.SocketsHttpHandlerPerfTest.Get(ssl: False, connectionPerRequest: True, chunkedResponse: True, responseLength: 1)
  [4/23/2018 1:31:47 PM][ERR] System.Net.Http.Tests.SocketsHttpHandlerPerfTest.Get(ssl: False, connectionPerRequest: True, chunkedResponse: True, responseLength: 1): Only one usage of each socket address (protocol/network address/port) is normally permitted
  [4/23/2018 1:31:47 PM][ERR]    at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken) in C:\J\w\perf_windows_---356c2fc4\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\ConnectHelper.cs:line 86
     at System.Net.Http.HttpConnectionPool.CreateConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken) in C:\J\w\perf_windows_---356c2fc4\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpConnectionPool.cs:line 342
     at System.Net.Http.HttpConnectionPool.WaitForCreatedConnectionAsync(ValueTask`1 creationTask) in C:\J\w\perf_windows_---356c2fc4\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpConnectionPool.cs:line 484
     at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken) in C:\J\w\perf_windows_---356c2fc4\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpConnectionPool.cs:line 275
     at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken) in C:\J\w\perf_windows_---356c2fc4\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\RedirectHandler.cs:line 32
     at System.Net.Http.Tests.SocketsHttpHandlerPerfTest.Get(Boolean ssl, Boolean connectionPerRequest, Boolean chunkedResponse, Int32 responseLength) in C:\J\w\perf_windows_---356c2fc4\src\System.Net.Http\tests\Performance\Perf.SocketsHttpHandler.cs:line 107
     at Microsoft.Xunit.Performance.BenchmarkTestInvoker.<>c__DisplayClass1_0.<<CallTestMethod>b__0>d.MoveNext()
  --- End of stack trace from previous location where exception was thrown ---
     at Microsoft.Xunit.Performance.Internal.BenchmarkIterator.RunAsync(Func`1 testMethod)
     at Xunit.Sdk.TestInvoker`1.<>c__DisplayClass46_1.<<InvokeTestMethodAsync>b__1>d.MoveNext() in C:\BuildAgent\work\cb37e9acf085d108\src\xunit.execution\Sdk\Frameworks\Runners\TestInvoker.cs:line 227
  --- End of stack trace from previous location where exception was thrown ---
     at Xunit.Sdk.ExecutionTimer.AggregateAsync(Func`1 asyncAction) in C:\BuildAgent\work\cb37e9acf085d108\src\xunit.execution\Sdk\Frameworks\ExecutionTimer.cs:line 48
     at Xunit.Sdk.ExceptionAggregator.RunAsync(Func`1 code) in C:\BuildAgent\work\cb37e9acf085d108\src\xunit.core\Sdk\ExceptionAggregator.cs:line 90
  [4/23/2018 1:31:47 PM][INF]   System.Net.Http.Tests.SocketsHttpHandlerPerfTest.Get(ssl: True, connectionPerRequest: False, chunkedResponse: True, responseLength: 100000)
  [4/23/2018 1:31:52 PM][INF] Finished 16 tests in 62.075s (5 failed, 0 skipped)
  [ERROR] Benchmark execution failed.
    System.Exception: 5 benchmark(s) failed to execute.
     at Microsoft.Xunit.Performance.Api.XunitPerformanceHarness.<RunBenchmarks>g__xUnitAction8_0(String assemblyPath)
     at Microsoft.Xunit.Performance.Api.ETWProfiler.Record(XUnitPerformanceSessionData xUnitPerformanceSessionData, XUnitPerformanceMetricData xUnitPerformanceMetricData, Action action)
     at Microsoft.Xunit.Performance.Api.XunitPerformanceHarness.RunBenchmarks(String assemblyFileName)
     at PerfHarness.Main(String[] args) in C:\J\w\perf_windows_---356c2fc4\src\Common\perf\PerfRunner\PerfRunner.cs:line 19

```

--- a/src/System.Net.Http/tests/Performance/Perf.SocketsHttpHandler.cs
+++ b/src/System.Net.Http/tests/Performance/Perf.SocketsHttpHandler.cs
@@ -22,14 +22,13 @@ namespace System.Net.Http.Tests
 
         public static IEnumerable<object[]> Get_MemberData() =>
             from ssl in new[] { false, true }
-            from connectionPerRequest in new[] { false, true }
             from chunkedResponse in new[] { false, true }
             from responseLength in new[] { 1, 100_000 }
-            select new object[] { ssl, connectionPerRequest, chunkedResponse, responseLength };
+            select new object[] { ssl, chunkedResponse, responseLength };
 
-        [Benchmark(InnerIterationCount = InnerIterationCount, Skip = "https://github.com/dotnet/corefx/issues/29308")]
+        [Benchmark(InnerIterationCount = InnerIterationCount)]
         [MemberData(nameof(Get_MemberData))]
-        public async Task Get(bool ssl, bool connectionPerRequest, bool chunkedResponse, int responseLength)
+        public async Task Get(bool ssl, bool chunkedResponse, int responseLength)
         {
             using (var serverCert = System.Net.Test.Common.Configuration.Certificates.GetServerCertificate())
             using (var listener = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
@@ -38,7 +37,7 @@ namespace System.Net.Http.Tests
                 listener.Bind(new IPEndPoint(IPAddress.Loopback, 0));
                 listener.Listen(int.MaxValue);
                 string responseText =
-                    "HTTP/1.1 200 OK\r\n" + (connectionPerRequest ? "Connection: close\r\n" : "") + (chunkedResponse ?
+                    "HTTP/1.1 200 OK\r\n" + (chunkedResponse ?
                     $"Transfer-Encoding: chunked\r\n\r\n{responseLength.ToString("X")}\r\n{new string('a', responseLength)}\r\n0\r\n\r\n" :
                     $"Content-Length: {responseLength}\r\n\r\n{new string('a', responseLength)}");
                 ReadOnlyMemory<byte> responseBytes = Encoding.UTF8.GetBytes(responseText);
@@ -67,10 +66,6 @@ namespace System.Net.Http.Tests
                                         {
                                             while (!string.IsNullOrEmpty(await reader.ReadLineAsync()));
                                             await stream.WriteAsync(responseBytes);
-                                            if (connectionPerRequest)
-                                            {
-                                                break;
-                                            }
                                         }
                                     }
                                 }
@@ -93,10 +88,6 @@ namespace System.Net.Http.Tests
                     }
 
                     var req = new HttpRequestMessage(HttpMethod.Get, uri);
-                    if (connectionPerRequest)
-                    {
-                        req.Headers.ConnectionClose = true;
-                    }
 
                     foreach (BenchmarkIteration iteration in Benchmark.Iterations)
                     {

